packages/blade/src/components/BottomSheet/_decisions/decisions.md

```diff
@@ -0,0 +1,276 @@
+| -------- | --------------- | ----------- | ------------------------------------ | -------- |
+| `title`  | `string`        | `undefined` | Title of the Header                    | ✅       |
+| `prefix` | `IconComponent` | `undefined` | Prefix icon placed before title text |          |
+
+### `BottomSheetHeaderTrailing` API
+
+| Prop     | Type                 | Default     | Description                                                | Required |
+| -------- | -------------------- | ----------- | ---------------------------------------------------------- | -------- |
+| `visual` | `Link, Action, Icon` | `undefined` | Trailing visual element placed on right side of the header |          |
+
+### `BottomSheetFooterLeading` API
+
+| Prop     | Type            | Default     | Description                          | Required |
+| -------- | --------------- | ----------- | ------------------------------------ | -------- |
+| `title`  | `string`        | `undefined` | Title of the BottomSheet footer      | ✅       |
+| `prefix` | `IconComponent` | `undefined` | Prefix icon placed before title text |          |
+
+### `BottomSheetFooterTrailing` API
+
+| Prop      | Type                                     | Default     | Description                                | Required |
+| --------- | ---------------------------------------- | ----------- | ------------------------------------------ | -------- |
+| `actions` | `{ primary: Action, secondary: Action }` | `undefined` | Renders a primary/secondary action buttons |          |
+
+```ts
+type Action = {
+  onClick: () => void;
+  text: string;
+  type?: 'button' | 'reset' | 'submit';
+  accessibilityLabel: string;
+  isLoading: boolean;
+  isDisabled: boolean;
+  icon: React.ReactNode;
+  iconPosition: 'left' | 'right';
+};
+```
+
+## Composition with Dropdown
+
+We will export `BottomSheet` component separately as an independant component but generally the pattern will be to use it with SelectInput, where in mobile devices the Select's dropdown will be replaced by the BottomSheet.
+
+There are two approaches to doing it:
+
+1. We coupled the BottomSheet & SelectInput tightly and internally conditionally switch the components
+
+Pros:
+
+- Easy implementation for user's end
+- User's don't have to think about breakpoints, conditional rendering etc
+- From blade side, we will have greater control over the pattern
+
+Cons:
+
+- Bundle size will be impacted, even if users are desktop they will get the bundle of BottomSheet (vice versa)
+
+2. We expose BottomSheet indepandantly and let users lazy load the component as needed
+
+Pros:
+
+- No uneccesary bundle size impact for any of the platforms
+
+Cons:
+
+- Not trivial to implement from user's end, they will have to compose the BottomSheet & Select as per their needs.
+
+
+Considering the bundle size downside to approach 1, we decided to go ahead with approach 2.
+
+### Composition Example
```

Comment:
This seems very tricky. The DX with approach 2 is quite confusing. How bad of an impact to the bundle size will we have if we do add bottom sheet in it as well?  If the impact isn't that heavy, we should really consider a better DX

Reply:
Need to test it out, at minimum it will have the bundle of the Dropdown 11kb + whatever the bottomsheet adds on top. 

Now considering the bundle impact and the [UX consideration](https://github.com/razorpay/blade/pull/1002#discussion_r1118614517) we can decide whether it's worth going for approach 2 or not.